File: JY/10월 4주차/행렬의곱셈.py
"""
[1 ,4]  [3, 3]
[3 ,2]  [3, 3]
[4 ,1]

3 x 2 
2 x 2  

= 3 x 2
for i -> A의 열
for j -> B의 열 또는 A의 행
sum = A[i, k] * B[k, j]
C[i][j] = sum
"""
def solution(arr1, arr2):
    # 3 X 5 * 5 X 2 = 3 X 2  
    # Y = A의 Y
    # X = B의 X 
    
    Y = len(arr1)
    X = len(arr2[0])
    add_count = len(arr1[0])
    # add_count = len(arr2)
    answer = [[0] * X  for _ in range(Y)] 
 
    for i in range(Y):
        for j in range(X):
            sum_value = 0
            for k in range(add_count):
                sum_value = sum_value +  (arr1[i][k] * arr2[k][j])
            answer[i][j] = sum_value
                
    return answer

File: DB/10월 3주차/행렬의곱셈.js
function solution(arr1, arr2) {
  let answer = [];

  //   배열에 미리 0 채우는 것이 계산에 더 효율적
  //   let answer = Array.from({ length: arr1.length }, () => Array(arr2[0].length).fill(0));

  for (let i = 0; i < arr1.length; i++) {
    let row = []; // 결과 행을 담을 배열
    for (let j = 0; j < arr2[0].length; j++) {
      let sum = 0; // 각 원소의 결과를 저장할 변수
      for (let k = 0; k < arr1[0].length; k++) {
        sum += arr1[i][k] * arr2[k][j];
      }
      row.push(sum); // 계산한 값을 현재 행에 추가
    }
    answer.push(row); // 완성된 행을 결과 배열에 추가
  }

  return answer;
}
